digitaltwins/samples/service/javascript/get_model.js

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+const IoTHubTokenCredentials = require('azure-iot-digitaltwins-service').IoTHubTokenCredentials;
+const DigitalTwinServiceClient = require('azure-iot-digitaltwins-service').DigitalTwinServiceClient;
+const { inspect } = require('util');
+
+// Simple example of how to:
+// - create a Digital Twin Service Client using the DigitalTwinServiceClient constructor
+// - get a Digital Twin Model by model ID
+//
+// Preconditions:
+// - Environment variables have to be set
+// - Twin enabled device must exist on the ADT hub
+async function main() {
+  const modelId = process.env.IOTHUB_MODEL_ID; // suggestion: urn:azureiot:Client:SDKInformation:1
+
+  // Create service client
+  const credentials = new IoTHubTokenCredentials(process.env.IOTHUB_CONNECTION_STRING);
+  const digitalTwinServiceClient = new DigitalTwinServiceClient(credentials);
+
+  // Get digital twin model
+  const model = await digitalTwinServiceClient.getModel(modelId);
```

Comment:
>getModel(modelId) [](start = 47, length = 17)

Interesting that these can existing independent of the context of a device.

Is there an API to push a model into a hub?

Reply:
No, there is no such API on IoTHub

---
In reply to: [450403979](https://github.com/Azure/azure-iot-sdk-node/pull/837#discussion_r450403979) [](ancestors = 450403979)